lib/spack/spack/test/cmd/config.py

```diff
@@ -399,7 +399,8 @@ def test_remove_list(mutable_empty_config):
 
```

Comment:
That is tested by `test_config_add_to_env` and related tests in `lib/spack/spack/test/cmd/config.py`. The test is old, but the code that is tested was ported over to take advantage of the new feature (because the old version was hacky).

Reply:
I think there should be an explicit test for comment preservation, as I mention in the thread related to that test.